R/pkg/R/sparkR.R

```diff
@@ -123,16 +123,30 @@ sparkR.init <- function(
     uriSep <- "////"
   }
 
+  sparkEnvirMap <- convertNamedListToEnv(sparkEnvir)
+
   existingPort <- Sys.getenv("EXISTING_SPARKR_BACKEND_PORT", "")
   if (existingPort != "") {
     backendPort <- existingPort
   } else {
     path <- tempfile(pattern = "backend_port")
+    submitOps <- Sys.getenv("SPARKR_SUBMIT_ARGS", "sparkr-shell")
+    # spark.driver.memory cannot be set in env:
+    # http://spark.apache.org/docs/latest/configuration.html#application-properties
+    # Add spark.driver.memory if set in sparkEnvir and not already set in SPARKR_SUBMIT_ARGS
+    if (!grepl("--driver-memory", submitOps)) {
```

Comment:
Lets add the support for other options of the same limitation, according to http://spark.apache.org/docs/latest/configuration.html, we also have:
spark.driver.extraClassPath
spark.driver.extraJavaOptions
spark.driver.extraLibraryPath